### PR TITLE
[Issue 9173] [Persistent Topics] Auto-create partitions even when the auto-creation is disabled

### PR DESCRIPTION
**Pull request by Jason918**
Related to #9173
Master Issue: #9173

### Motivation

Please see #9173

### Modifications

When the BrokerService is about to create a persistent topic it checks if it is a partition of an existing partitioned topic and if so, even the auto-creation is disabled, it creates it.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

An integration test is added assuring the new behavior.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by Jason918** [comment]
@codelipenghui @jiazhai Hi, can you please take a look?

**Comment by Anonymitaet** [comment]
Hi @Jason918 thanks for your contribution. For this PR, do we need to update docs?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 1]
```diff
@@ -863,8 +863,19 @@ public void unloadNamespaceBundlesGracefully() {
...
             final boolean isPersistentTopic = TopicName.get(topic).getDomain().equals(TopicDomain.persistent);
             if (isPersistentTopic) {
-                return topics.computeIfAbsent(topic, (topicName) -> {
-                    return this.loadOrCreatePersistentTopic(topicName, createIfMissing);
+                return topics.computeIfAbsent(topic, (name) -> {
+                    final TopicName topicName = TopicName.get(name);
+                    if (!createIfMissing && topicName.isPartitioned()) {
+                        final TopicName partitionedTopicName = TopicName.get(topicName.getPartitionedTopicName());
+                        return this.fetchPartitionedTopicMetadataAsync(partitionedTopicName).thenCompose((metadata) -> {
+                            if (topicName.getPartitionIndex() < metadata.partitions) {
```
Comment (gaoran10) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:871]: The `topicName.getPartitionIndex()` shouldn't be greater or equals with `metadata.partitions`, maybe this case should be rejected, we could throw an exception directly.

**Comment by codelipenghui** [comment]
Sorry Jason, after discussed, the correct way to resolve this issue is waiting for creating partitions when create the partitioned topic and users should enabled the topic auto-creation if they want to delete the inactive topic automatically. So I will close the PR first and both the issue https://github.com/apache/pulsar/issues/9173 first.

Feel free to reopen it if needed.

Thank.

**Pull request closed without merging**